docs/HowToConnectExternally.md

```diff
@@ -0,0 +1,298 @@
+For more information on how to customize the provisioned NLB, check [Network Load Balancer](https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/guide/service/nlb/).
+When the `internalIP` option is set to “true”, no additional configuration is needed.
+
+If the [legacy AWS cloud provider load balancer controller](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) is used the Kubernetes service annotations must be changed. For example:
+
+```yaml
+cloud:
+  provider: aws
+
+admin:
+  externalAccess:
+    enabled: true
+    internalIP: false
+    annotations:
+      service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+
+database:
+  te:
+    externalAccess:
+      enabled: true
+      internalIP: false
+      annotations:
+        service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+```
+
+### Azure
+
+No additional configuration is needed to enable NuoDB database external access in Azure Kubernetes Service (AKS).
+For more information, see [Use a public Standard Load Balancer](https://docs.microsoft.com/en-us/azure/aks/load-balancer-standard).
+
+## Troubleshooting
+
+### Unable to connect
+
+There may be different reasons for client connectivity problems such as:
+
+- not _Ready_ NuoDB Admin Pods
+- not _Ready_ TE pods
+- incorrect external access configuration
+- incorrect NuoDB Load Balancer configuration
+- incorrect NLB configuration
+- network connectivity problems including lack of routing, firewall configuration, and many more
+
+> **ACTION**: You can rule out any of the points above one by one.
+Start by checking the overall Pod status for the NuoDB domain and database.
+Some of the common troubleshooting steps are listed below, however, there might be additional verifications specific to your deployment.
+
+1. Verify that all AP, TE, and SM pods are reported _Ready_ using `kubectl get pods`.
+2. Check the NuoDB domain and database using `nuocmd show domain`.
+3. Verify the database availability inside the cluster is using the same connection properties as the application uses.
+To verify this, use the  `nuosql` tool inside an AP Pod.
+4. Make sure that `external-address` and/or `external-port` process labels are configured correctly on the TE database processes using `nuocmd --show-json-fields hostname,labels get processes`.
+If you are using the `--enable-external-access` process option, verify that the configured values are the same as the `EXTERNAL-IP` shown for the Kubernetes service in `kubectl get services` output.
+If the value is not the same, restart the TE database process and verify again.
+5. Verify that the `LBQuery` or `LBPolicy` syntax is correct. If using `LBPolicy`, verify that the expected policies are configured in the NuoDB Admin using `nuocmd get load-balancers` and `nuocmd get load-balancer-config`.
```

Comment:
Another debugging tool that can be used to test `LBQuery` syntax is the `nuocmd get sql-connection` subcommand. I don't remember if a bogus `LBQuery` would actually generate a clear error message, or if the user would have to dig through `nuoadmin.log `(`nuoadmin` stdout).

Reply:
Good point!
I think it good enough for testing the LBQuery syntax:

```
bash-4.4$ nuocmd get sql-connection --db-name test --connection-properties LBQuery "random(labels(dummy foo))"
'get sql-connection' failed: Unable to get a SQL connection for database test: Next token does not match " *(\bcolocated\b|\baddress\b|\bor\b|\bstart_id\b|\bpid\b|\blabel\b|\bserver_id\b|\bany\b|\bhostname\b|\bnot\b|\band\b|\bregion\b|\bfirst\b|\bnode_id\b|\boption\b) *": random(|labels(dummy foo))

bash-4.4$ nuocmd get sql-connection --db-name test --connection-properties LBQuery "random(label(dummy foo))"

bash-4.4$ nuocmd get sql-connection --db-name test --connection-properties LBQuery "random(label(te te1))"
Process(address=test-te-1:48006, ...
```